### PR TITLE
Add E0191, E0220 and E0232 error explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #24407.
r? @Manishearth

### CODE REVIEW

File: src/librustc_typeck/diagnostics.rs [hunk 1]
```diff
@@ -1550,6 +1550,30 @@ impl Foo for Bar {
 ```
 "##,
 
+E0191: r##"
```
Comment (arielb1) [inline src/librustc_typeck/diagnostics.rs]: You may want to make the reason clearer (trait objects need to have all associated types specified).

File: src/librustc_typeck/diagnostics.rs [hunk 2]
```diff
@@ -1831,6 +1855,31 @@ extern "rust-intrinsic" {
...
+```
+trait Trait {
+    type Bar;
+}
+
+type Foo = Trait<F=i32>; // error: associated type `F` not found for
+                         //        `Trait`
+```
+
+Please verify you used the good trait or you didn't mispelled the
```
Comment (arielb1) [inline src/librustc_typeck/diagnostics.rs]: *right trait

**Comment by GuillaumeGomez** [comment]
@arielb1: Updated.

File: src/librustc_typeck/diagnostics.rs [hunk 3]
```diff
@@ -1550,6 +1550,31 @@ impl Foo for Bar {
...
+```
+trait Trait {
+    type Bar;
+}
+
+type Foo = Trait; // error: the value of the associated type `Bar` (from
+                  //        the trait `Trait`) must be specified
+```
+
+Please verify you specified all associated types of the trait or that you
```
Comment (arielb1) [inline src/librustc_typeck/diagnostics.rs]: *and

File: src/librustc_typeck/diagnostics.rs [hunk 4]
```diff
@@ -1831,6 +1856,47 @@ extern "rust-intrinsic" {
...
+
+E0232: r##"
+The attribute must have a value. Erroneous code example:
+
+```
+#[rustc_on_unimplemented] // error: this attribute must have a value
+trait Bar {}
+```
+
+Please inquire the missing value of the attribute. Example:
```
Comment (arielb1) [inline src/librustc_typeck/diagnostics.rs]: s/inquire/supply/

**Comment by arielb1** [comment]
r=me modulo grammar errors

**Comment by GuillaumeGomez** [comment]
Squashed.

**Pull request merged**